src/main/java/it/gov/pagopa/payhub/activities/service/treasury/TreasuryOpi14MapperService.java

```diff
@@ -0,0 +1,73 @@
+package it.gov.pagopa.payhub.activities.service.treasury;
+
+import it.gov.pagopa.payhub.activities.dto.IngestionFlowFileDTO;
+import it.gov.pagopa.payhub.activities.dto.OrganizationDTO;
+import it.gov.pagopa.payhub.activities.dto.treasury.TreasuryDTO;
+import it.gov.pagopa.payhub.activities.enums.TreasuryOperationEnum;
+import it.gov.pagopa.payhub.activities.util.TreasuryUtils;
+import it.gov.pagopa.payhub.activities.xsd.treasury.opi14.FlussoGiornaleDiCassa;
+import it.gov.pagopa.payhub.activities.xsd.treasury.opi14.InformazioniContoEvidenza;
+
+import java.util.*;
+
+public class TreasuryOpi14MapperService implements TreasuryMapperService<FlussoGiornaleDiCassa, Map<TreasuryOperationEnum, List<TreasuryDTO>>> {
+    @Override
+    public Map<TreasuryOperationEnum, List<TreasuryDTO>> apply(FlussoGiornaleDiCassa fGC, IngestionFlowFileDTO ingestionFlowFileDTO) {
+        Map<TreasuryOperationEnum, List<TreasuryDTO>> resultMap = new EnumMap<>(TreasuryOperationEnum.class);
```

Comment:
The grouping Logic could be implemented once in the strategy class.
Moreover, this could be written cleaner and more optimized using collection's stream api and groupingBy collector